cli/remotebazel/remotebazel.go

```diff
@@ -779,17 +795,20 @@ func Run(ctx context.Context, opts RunOpts, repoConfig *RepoConfig) (int, error)
 	log.Debugf("Invocation ID: %s", iid)
 
 	// If the remote bazel process is canceled or killed, cancel the remote run
-	sigChan := make(chan os.Signal, 1)
 	go func() {
-		<-sigChan
-		_, err = bbClient.CancelExecutions(ctx, &inpb.CancelExecutionsRequest{
+		<-ctx.Done()
+		// Use a non-cancelled context to ensure the remote executions are
+		// canceled
+		nonCanceledCtx := context.Background()
+		nonCanceledCtx = metadata.AppendToOutgoingContext(nonCanceledCtx, "x-buildbuddy-api-key", opts.APIKey)
+		_, err = bbClient.CancelExecutions(nonCanceledCtx, &inpb.CancelExecutionsRequest{
```

Comment:
as of go 1.21 there's context.WithoutCancel for this https://pkg.go.dev/context#WithoutCancel

Reply:
Oh nice - updated